siliconcompiler/schema/schema_obj.py

```diff
@@ -462,6 +473,9 @@ def _normalize_field(value, sc_type, field, keypath):
         def error_msg(t):
             return f'Invalid value {value} for field {field} of keypath {keypath}: expected {t}'
 
+        if field in ('author', 'filehash', 'date', 'hashalgo', 'copy') and ('file' not in sc_type):
```

Comment:
This field check if almost the same as PERNODE_FIELDS, would it be possible to use that here?

Reply:
These are fields that are valid only for file-type values, which happen to overlap a bit but are distinct meaning-wise.